neo/app/controllers/Application.scala

```diff
@@ -83,10 +86,87 @@ class Application @Inject()(val wsClient: WSClient,
     }
   }
 
+  def getCity(NELat: Double, SWLat: Double, NELng: Double, SWLng: Double) = Action{
+    Ok(Application._getCity(NELat, SWLat, NELng, SWLng, cities))
+  }
+
   class Listener extends Actor {
     def receive = {
       case d: DeadLetter => println(d)
     }
   }
 
 }
+
+object Application{
+  def loadCity(file: File): List[JsValue] = {
+    val stream = new FileInputStream(file)
+    val json = Json.parse(stream)
+    val features = "features"
+    val geometry = "geometry"
+    val type_str = "type"
+    val coordinates = "coordinates"
+    val Polygon = "Polygon"
+    val MultiPolyton = "MultiPolygon"
+    val values = (json \ features).as[List[JsObject]]
+    val newValues = List.newBuilder[JsValue]
+    for(n <- values.indices) {
+      val thisValue = values.apply(n)
+      val geoType = (thisValue \ geometry \ type_str).as[String]
+
+      geoType match{
+        case Polygon => {
+          val corr  = (thisValue \ geometry \ coordinates).as[JsArray].apply(0).as[List[List[Double]]]
+          val Xcorr = corr.map( x => x.apply(0))
```

Comment:
these code
```
          val Xcorr = corr.map( x => x.apply(0))
           val Ycorr = corr.map( x => x.apply(1))
           val minX = Xcorr.reduceLeft((x, y) => if (x < y) x else y)
           val maxX = Xcorr.reduceLeft((x, y) => if (x > y) x else y)
           val minY = Ycorr.reduceLeft((x, y) => if (x < y) x else y)
           val maxY = Ycorr.reduceLeft((x, y) => if (x > y) x else y)
```
can be written as 
```
val (minX, maxX, minY, maxY) = corr.reduceLeft(180,-180,180,-180) { case ((minX, maxX, minY, maxY), e) => 
 ( math.min(minX, e(0)),  math.max(maxX, e(0)),  math.min(minY, e(1)),  math.max(minY, e(1)))
}
```
In this way, you can get 4 values in one loop and avoid creating 2 arrays(x and y).

P.S., try to use full name, `corr` -> `coordinate`. And using the plural form if it's an array. e.g., in this case, `corr` is an array, it should be `coordinates`.

PPS, the X and Y is easy to make mistake as for coordinates, `latitude` and `longitude` make more sense.